kernel/src/scan/data_skipping.rs

```diff
@@ -188,11 +189,7 @@ impl DataSkippingFilter {
             static ref FILTER_EXPR: Expr = Expr::column("predicate").distinct(Expr::literal(false));
         );
 
-        let predicate = match predicate {
-            Some(predicate) => predicate,
-            None => return None,
-        };
-
+        let predicate = predicate.as_deref()?;
```

Comment:
Because Try isn't implemented on `&Option` as_deref converts `&Option<T>` to `Option<&T>` https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref